src/user.ts

```diff
@@ -211,20 +231,48 @@ export class User extends AsyncCreatable<User.Options> {
     super(options);
     this.org = options.org;
   }
+
   /**
    * Generate default password for a user. Returns An encrypted buffer containing a utf8 encoded password.
    */
-  public static generatePasswordUtf8(): SecureBuffer<void> {
-    // one character of each of the 4 types followed by the remaining random characters, then quasi-randomize
+  public static generatePasswordUtf8(
+    passwordCondition: PasswordConditions = { length: 10, complexity: 1 }
+  ): SecureBuffer<void> {
     const ALLCHARS = UPPER.concat(LOWER, NUMBERS, SYMBOLS.join(''));
-    const password = [UPPER[rand(UPPER)], LOWER[rand(LOWER)], NUMBERS[rand(NUMBERS)], SYMBOLS[rand(SYMBOLS)]]
-      .concat(
-        Array(PASSWORD_LENGTH - 4)
+    let password: string[] = [];
+
+    // Required number of Lower letter characters. one character of each of the 4 types followed by the remaining random characters, then quasi-randomize
```

Comment:
overall, this feels like it could be simpler with booleans.  They all require 1 or 0, nothing higher than 1.

So, for example
```js
if (PASSWORD_COMPLEXITY[passwordCondition.complexity].LOWER) {
    password.push(LOWER[rand(LOWER)])
}
```

Reply:
another idea would be to iterate the types to avoid the repetition, something like

```ts
[SYMBOLS,NUMBERS,UPPER,LOWER].forEach( charSet => {
    if (PASSWORD_COMPLEXITY[passwordCondition.complexity][charSet]) {
        password.push(charType[rand(charSet)])
    }
})
```

Reply:
finally, we could generate the right length of password from the start (`length` randow lowercase characters) then overwrite it with whatever other characters are requied and then shuffle that.

Then there's a chance (for cases 0 and 1) where no changes have to be made at all (all the requirements were false)